### PR TITLE
scheduler: add operator influence when sort store

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve? 

After #1823 , we changed balance region scheduler, but we didn't consider operator influence when sort store.

### What is changed and how it works?

Add operator influence when sort store

- [x] Test.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by lhy1024** [comment]
It work on the state that all store are the same and are in the second function
image

But it doesn't work on the state that to transition from the first function to the second function. #1795 will improve the ability for transition.

image

File: pkg/mock/mockcluster/mockcluster.go [hunk 1]
```diff
@@ -216,6 +216,19 @@ func (mc *Cluster) AddRegionStore(storeID uint64, regionCount int) {
 	mc.PutStore(store)
 }
 
+// AddRegionStoreWithLeader adds store with specified count of region and leader.
+func (mc *Cluster) AddRegionStoreWithLeader(storeID uint64, regionCount int, leaderCounts ...int) {
```
Comment (rleungx) [inline pkg/mock/mockcluster/mockcluster.go]: Can we use `AddRegionStore` and `UpdateLeaderCount` to do it or some existing methods?

File: server/schedulers/balance_test.go [hunk 2]
```diff
@@ -663,7 +663,7 @@ func (s *testBalanceRegionSchedulerSuite) checkReplica5(c *C, tc *mockcluster.Cl
 	testutil.CheckTransferPeer(c, sb.Schedule(tc)[0], operator.OpBalance, 11, 6)
 }
 
-// TestBalance2 for cornor case 1:
+// TestBalance2 for cover case 1:
```
Comment (rleungx) [inline server/schedulers/balance_test.go]: I guess it should be "corner".

File: server/schedulers/balance_test.go [hunk 3]
```diff
@@ -757,6 +757,33 @@ func (s *testBalanceRegionSchedulerSuite) TestReplacePendingRegion(c *C) {
...
+	oc := schedule.NewOperatorController(s.ctx, tc, mockhbstream.NewHeartbeatStream())
+	sb, err := schedule.CreateScheduler(BalanceRegionType, oc, core.NewStorage(kv.NewMemoryKV()), schedule.ConfigSliceDecoder(BalanceRegionType, []string{"", ""}))
+	c.Assert(err, IsNil)
+	opt.SetMaxReplicas(1)
+	// Add stores 1,2,3,4.
+	tc.AddRegionStoreWithLeader(1, 2)
+	tc.AddRegionStoreWithLeader(2, 8)
+	tc.AddRegionStoreWithLeader(3, 8)
+	tc.AddRegionStoreWithLeader(4, 16, 8)
+	// add 8 leader region to store 4 and move them to store 3
```
Comment (disksing) [inline server/schedulers/balance_test.go]: 8 leader regions

File: pkg/mock/mockcluster/mockcluster.go (hunk 1 above)
Comment (lhy1024) [reply pkg/mock/mockcluster/mockcluster.go]: the two method only can update region num rather than adding region

**Comment by lhy1024** [comment]
@nolouch @rleungx PTAL

File: server/schedulers/balance_test.go [hunk 4]
```diff
@@ -153,6 +153,27 @@ func (s *testBalanceSpeedSuite) TestBalanceLimit(c *C) {
 	c.Assert(adjustBalanceLimit(tc, core.LeaderKind), Equals, uint64(math.Sqrt(50.0/2.0)))
 }
 
+func (s *testBalanceSuite) TestTolerantRatio(c *C) {
+	opt := mockoption.NewScheduleOptions()
+	tc := mockcluster.NewCluster(opt)
+	// create a region to control average region size.
+	tc.AddLeaderRegion(1, 1, 2)
```
Comment (rleungx) [inline server/schedulers/balance_test.go]: check the returned value?

Comment (nolouch) [reply server/schedulers/balance_test.go]: PTAL @lhy1024

**Pull request merged**